src/network.rs

```diff
@@ -54,6 +54,15 @@ pub(crate) enum NetworkDataInner<H: Hasher, D: Data, S: Signature, MS: PartialMu
     Alert(AlertMessage<H, D, S, MS>),
 }
 
+impl<H: Hasher, D: Data, S: Signature, MS: PartialMultisignature> NetworkDataInner<H, D, S, MS> {
```

Comment:
BTW, is there any thought behind having a separate inner type?

Reply:
The type that is `Encode`/`Decode` has to be public and we don't want the internal representation of network data (so the enum types `Alert` and `Units` in our case) to be part of our API.

Reply:
Makes sense